client/my-sites/themes/themes-selection.jsx

```diff
@@ -168,15 +168,20 @@ const ThemesSelection = React.createClass( {
 } );
 
 const ConnectedThemesSelection = connect(
-	( state, { filter, page, search, tier, vertical, siteId, queryWpcom } ) => {
+	( state, { filter, page, search, tier, vertical, siteId, source } ) => {
 		const isJetpack = isJetpackSite( state, siteId );
-		const siteIdOrWpcom = ( siteId && isJetpack && ! ( queryWpcom === true ) ) ? siteId : 'wpcom';
+		let siteIdOrWpcom;
+		if ( source === 'wpcom' || source === 'wporg' ) {
+			siteIdOrWpcom = source;
```

Comment:
Hmm, then again, that would require us to pass down yet another prop through `ThemeShowcase` because of our 💩ty component hierarchy. So maybe that's something for that PR that's going to refactor that hierarchy. I'll see if I can just do a simple rename here.